exchanges/deribit/deribit_websocket.go

```diff
@@ -698,51 +702,59 @@ func (d *Deribit) processOrderbook(respRaw []byte, channels []string) error {
 	}
 	var assetType asset.Item
 	if len(channels) == 3 {
-		cp, err := currency.NewPairFromString(channels[1])
+		cp, err := currency.NewPairFromString(orderbookData.InstrumentName)
 		if err != nil {
 			return err
 		}
-		asks := make(orderbook.Items, len(orderbookData.Asks))
-		for x := range asks {
+		asks := make(orderbook.Items, 0, len(orderbookData.Asks))
+		for x := range orderbookData.Asks {
 			if len(orderbookData.Asks[x]) != 3 {
 				return errMalformedData
 			}
 			price, okay := orderbookData.Asks[x][1].(float64)
 			if !okay {
 				return fmt.Errorf("%w, invalid orderbook price", errMalformedData)
+			} else if price == 0 {
+				continue
 			}
 			amount, okay := orderbookData.Asks[x][2].(float64)
 			if !okay {
 				return fmt.Errorf("%w, invalid amount", errMalformedData)
 			}
-			asks[x] = orderbook.Item{
+			asks = append(asks, orderbook.Item{
 				Price:  price,
 				Amount: amount,
-			}
+			})
 		}
-		bids := make([]orderbook.Item, len(orderbookData.Bids))
-		for x := range bids {
+		bids := make([]orderbook.Item, 0, len(orderbookData.Bids))
+		for x := range orderbookData.Bids {
 			if len(orderbookData.Bids[x]) != 3 {
 				return errMalformedData
 			}
 			price, okay := orderbookData.Bids[x][1].(float64)
 			if !okay {
 				return fmt.Errorf("%w, invalid orderbook price", errMalformedData)
+			} else if price == 0.0 {
+				continue
```

Comment:
This seems to only be a bid issue because group is not set as per [these docs](https://docs.deribit.com/#book-instrument_name-group-depth-interval) and what it does is round to the whole number. This is kind of important to keep pricing to what the raw data is for arbitrage purposes so if you can please set group to `none` that would be the best. You can see from this picture of descending tranche levels it loses its value right at the end, the price is < 1 and remove these filters.

<img width="1897" alt="image" src="https://github.com/thrasher-corp/gocryptotrader/assets/9391715/47ea77dc-83f4-40d1-83e9-eaf19cefb31b">